core/sail/fts/lucene-spin/src/main/java/org/eclipse/rdf4j/lucene/spin/LuceneSpinSailConnection.java

```diff
@@ -0,0 +1,261 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Eclipse RDF4J contributors, Aduna, and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.lucene.spin;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.Literal;
+import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.sail.NotifyingSailConnection;
+import org.eclipse.rdf4j.sail.SailConnectionListener;
+import org.eclipse.rdf4j.sail.SailException;
+import org.eclipse.rdf4j.sail.helpers.NotifyingSailConnectionWrapper;
+import org.eclipse.rdf4j.sail.lucene.LuceneSailBuffer.AddRemoveOperation;
+import org.eclipse.rdf4j.sail.lucene.LuceneSailBuffer.ClearContextOperation;
+import org.eclipse.rdf4j.sail.lucene.LuceneSailBuffer.Operation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.eclipse.rdf4j.sail.lucene.LuceneSailBuffer;
+import org.eclipse.rdf4j.sail.lucene.LuceneSailConnection;
+import org.eclipse.rdf4j.sail.lucene.SearchIndex;
+
+/**
+ * This connection inherits Lucene index supporting methods from {@link LuceneSailConnection}.
+ * 
+ * @author sauermann
+ * @author christian.huetter
+ * @author github.com/jgrzebyta
+ */
+public class LuceneSpinSailConnection extends NotifyingSailConnectionWrapper {
+
+	private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+	private final SearchIndex luceneIndex;
+
+	/**
+	 * the buffer that collects operations
+	 */
+	final private LuceneSailBuffer buffer = new LuceneSailBuffer();
+
+	/**
+	 * The listener that listens to the underlying connection. It is disabled during clearContext operations.
+	 */
+	protected final SailConnectionListener connectionListener = new SailConnectionListener() {
+
+		@Override
+		public void statementAdded(Statement statement) {
+			// we only consider statements that contain literals
+			if (statement.getObject() instanceof Literal) {
+				if (statement == null)
```

Comment:
+1, this method needs to be reformatted and calls moved around to fit with RDF4J conventions